### PR TITLE
Replace `tsup` with `rollup` for bundling

### PR DESCRIPTION
**Pull request by rygine**
in this PR:

### replaced `tsup` with `rollup`

`tsup` does not generate accurate source maps when tree-shaking. it also seems like `tsup` isn't updated regularly even though there are several issues. with `rollup` we have more control of the bundling, accurate source maps, and smaller bundle sizes.

### removed CommonJS bundle output

the CJS bundle is mostly useless in a browser environment, so it's been removed. this is a breaking change and will require a major version bump.

### removed read receipt content type from React SDK

the read receipt content type has been flagged for review and is likely to change. as a result, we're removing official support for it at this time.

### CODE REVIEW

**Comment by neekolas** [comment]
Curious why rollup and not one of the newer shinier things like `esbuild`

**Comment by rygine** [comment]
> Curious why rollup and not one of the newer shinier things like `esbuild`

unfortunately, `esbuild` is the reason for the switch. it has issues with tree-shaking and `tsup` hands off the `esbuild` output to `rollup` for more processing, which leads to bad source maps.

with `rollup`, we can still use `esbuild` with a plugin, but i don't think any speed increases are worth the extra complexity, at least not for our SDKs. the current build only takes a few seconds.

`rollup` is quite shiny and is under heavy active development. the v4 release (late last year) changed the parser to `swc` for performance improvements. it has a great ecosystem as well with tons of plugins.

File: packages/react-sdk/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        declarationMap: false,
+      }),
+      terser(),
+      filesize({
+        showMinifiedSize: false,
+      }),
+    ],
+    external: [
+      "@xmtp/content-type-reaction",
+      "@xmtp/content-type-read-receipt",
```
Comment (daria-github) [inline packages/react-sdk/rollup.config.js:29]: intentional?

Comment (rygine) [reply packages/react-sdk/rollup.config.js:29]: since we're still exporting the config for this content type, it's in this list.

**Comment by neekolas** [comment]
@rygine you sold me. Mostly was just asking out of curiosity. Hadn't heard the name "rollup" in quite a while.

**Review by neekolas — APPROVED** [review]

**Comment by rygine** [comment]
> @rygine you sold me. Mostly was just asking out of curiosity. Hadn't heard the name "rollup" in quite a while.

i thought the same thing when i first set out to find a replacement for `tsup`. i was pleasantly surprised to discover how much work has been put into `rollup` recently.

**Pull request merged**